### PR TITLE
#515 switched to SVG icons, #525 added Ko-Fi platform

### PR DESCRIPTION
**Pull request by SeaLife**

### CODE REVIEW

File: webroot/js/components/platform-logos-list.js [hunk 1]
```diff
@@ -1,10 +1,52 @@
...
+  const platformInfo = SOCIAL_PLATFORMS[platform.toLowerCase()];
+  const inList = !!platformInfo;
+  const iconSupplied = !!icon
+  const name = inList ? platformInfo.name : platform;
+  const colorSupplied = !!color
+
+  const finalIcon = iconSupplied ? icon : (inList ? platformInfo.icon : '/img/socialhandles/default.svg')
+  const finalColor = colorSupplied ? color : (inList ? platformInfo?.color ? platformInfo.color : '#000000' : '#000000');
+
+  const style = `-webkit-mask-image: ${finalIcon}; mask-image: url(${finalIcon}); background-color: ${finalColor};`
```
Comment (gingervitis) [inline webroot/js/components/platform-logos-list.js:19]: Hey @SeaLife I tried checking out your SVG icons, but I couldn't get them to show up. For some reason the `mask-image` styles aren't getting rendered for me.

image

**Comment by gingervitis** [comment]
Thanks for working on converting some logos to SVG! 
I'm currently working on building out administrating the config through the Admin site, and I was thinking instead of keeping a current JS object of icons I was considering having the admin just read all the available icon files from a folder (them being svg format or gif or png or whatever). Could we just define the icon's color directly in svg code? Like just add the fill color or stroke color as attributes to the xml. like `<path fill="#1DA1F2" ...>` for Twitter.

**Comment by SeaLife** [comment]
Hey, this is probably possible @gingervitis, im not that deep into SVGs, so i got my SVG files from https://simpleicons.org/ and they are all color-less, but the required color is stated there.

You can ofc. modify/change this. (or i could do this, probably)

**Comment by SeaLife** [comment]
@gingervitis i've added the suggested change regarding the colors. Its still working for me :)
I need to remove some left-overs in the go-code and you should be ready to go :)

**Comment by gabek** [comment]
Looks ok to me when I just tested it.  Curious if it works for ginger, as she had issues with them displaying properly before.

**Comment by SeaLife** [comment]
In my last commit i changed the classes, because i set the color for the icons in the SVG files now, so @gingervitis should re-test this :)

**Comment by gabek** [comment]
Tested and looks good on my end.

**Comment by gingervitis** [comment]
Hey sorry finally got around to looking at it! Seems to be working for me now. Looks good.

**Pull request merged**